micro_dl/cli/curator_script.py

```diff
@@ -0,0 +1,167 @@
+            try:
+                assert len(PosList) > out_shape[0]
+            except AssertionError:
+                print(
+                    "number of positions listed in config exceeds number of positions in dataset"
+                )
+            pos = int(position.split("/")[2])
+            for gt_chan in ground_truth_chans:
+                if pos in PosList:
+                    target_data = im[0, chan_names.index(gt_chan), ...]
+                    Z, Y, X = target_data.shape
+                    focus_idx_target = z_list[pos]
+                    # focus_idx_target = focus_from_transverse_band(
+                    #     target_data, NA_det, lambda_illu, pxl_sz
+                    # )
+                    target_focus_slice = target_data[
+                        focus_idx_target, :, :
+                    ]  # FL focus slice image
+
+                    im_target = Image.fromarray(
+                        target_focus_slice
+                    )  # save focus slice as single page tif
+                    save_name = (
+                        "_p" + str(format(pos, "03d")) + "_z" + str(focus_idx_target)
+                    )
+                    im_target.save(
+                        os.path.join(
+                            target_zarr_dir,
+                            ground_truth_subdir,
+                            gt_chan + save_name + ".tif",
+                        )
+                    )
+
+                    source_focus_slice = im[
+                        0, chan_names.index(labelFree_chan[0]), focus_idx_target, :, :
+                    ]  # lable-free focus slice image
+                    im_source = Image.fromarray(
+                        source_focus_slice
+                    )  # save focus slice as single page tif
+                    im_source.save(
+                        os.path.join(
+                            target_zarr_dir,
+                            ground_truth_subdir,
+                            labelFree_chan[0] + save_name + ".tif",
+                        )
+                    )  # save for reference
+
+                    cp_mask = metrics.cpmask_array(
+                        target_focus_slice, cp_model
+                    )  # cellpose segmnetation for binary mask
+                    iio.imwrite(
+                        os.path.join(
+                            target_zarr_dir,
+                            ground_truth_subdir,
+                            gt_chan + save_name + "_cp_mask.png",
+                        ),
+                        cp_mask,
+                    )  # save binary mask as numpy or png
+
+    # segment prediction and add mask as channel to pred_dir
+    pred_plate = ngff.open_ome_zarr(store_path=pred_dir, mode="r+")
+    # im_pred = pred_plate.data
+    chan_names = pred_plate.channel_names
+
+    for channel_name in chan_names:
+        for i, (position, pos_data) in enumerate(pred_plate.positions()):
```

Comment:
No need for `enumerate()` if the index is not used.

Also, it will be more efficient iterate over positions in the outer loop, since opening positions are more expensive.

Reply:
I have removed the unused enumeration in two places